### ISSUE TITLE
feature: support JSON(typed_dict=TypedDict() ) Input, Output

### ISSUE THREAD

**Issue by KimSoungRyoul** [bentoml/BentoML#3748]
### to be 
~~~Python

class IAmSchema(BaseModel):
    a: list[float]
    b: list[float]
    c: list[float]
    d: Optional[float] # float | None



@svc.api(
    input=JSON( pydantic_model=IAmSchema ),
    output=JSON.from_sample({"qqqq":"wwww"}), 
)
async def predict_json(i_am_schema: ASchema) -> Dict[str,Any]:
    result = await runner.async_run(torch.tensor(list(i_am_schema.dict().values())))

    return {"qqqq": result}

~~~

### as is

~~~Python

class IAmSchema(t.TypedDict, total=False):
    a: list[float]
    b: list[float]
    c: list[float]
    d: NotRequried[float] # use NotRequired instead of Optional
    # Optional is rejected Spec in TypedDict (pep-0655)
    #  https://peps.python.org/pep-0655/#special-syntax-around-the-key-of-a-typeddict-item


@svc.api(
    input=JSON( 
        typeddict=IAmSchema, # <--- use TypedDict
    ),
    output=JSON.from_sample({"qqqq":"wwww"})), # <-- JSON IODescriptor cann't specify specifications in OAS3.
)
async def predict_json(i_am_schema: IAmSchema) -> Dict[str, Ant]:
     result = await runner.async_run(torch.tensor(list(i_am_schema.values()))

    return {"qqqq2222": result}


~~~

### Motivation

If we use Pydantic just to specify a type schema without validation, the logic of `xx.dict()` -> `Pydantic.parse_obj()` unnecessary layer which can affect performance. In this case, using TypedDict seems necessary.

~here is POC https://github.com/bentoml/BentoML/pull/3749~
https://github.com/bentoml/BentoML/pull/3809

I'd like to hear your thoughts on TypedDict

**Comment by aarnphm** [comment]
If we want to support TypedDict then it should just be supported by JSON and using the typing system, to some extend (but parsing types are costly, esp for IO descriptor as it will affect users throughput). 

TypedDict is only available from 3.8+ so this means we will have to depends on typing_extensions.

From a quick glance from the implementation, it seems like you are just using Pydantic, so I don't see the point of having this as a part of the core library.

The core usage of Pydantic is for validation. So if we want to support TypedDict and attrs class to an extend, it should probably be under the JSON IO descriptor.

**Comment by KimSoungRyoul** [comment]
This issue seems to be pointless in the current 1.2 version. 
and 
if need necessary, a new discussion issue should be created in  version 1.2.
